feature/experimental/p4rt/otg_tests/traceroute_packetout_test/traceroute_packetout_test.go

```diff
@@ -264,22 +244,100 @@ func TestPacketOut(t *testing.T) {
 		t.Fatalf("Couldn't parse router MAC: %v", err)
 	}
 
-	args := &testArgs{
-		ctx:    ctx,
-		leader: leader,
-		dut:    dut,
-		ate:    ate,
-		top:    top,
-		srcMAC: srcMAC,
-		dstMAC: dstMAC,
+	if err := setupP4RTClient(ctx, client); err != nil {
+		t.Fatalf("Could not setup p4rt client: %v", err)
 	}
 
-	if err := setupP4RTClient(ctx, args); err != nil {
-		t.Fatalf("Could not setup p4rt client: %v", err)
+	for _, m := range genMetadataCombinations() {
+		for _, ipv6 := range []bool{false, true} {
+			args := &testArgs{
+				ctx:         ctx,
+				client:      client,
+				dut:         dut,
+				ate:         ate,
+				top:         top,
+				srcMAC:      srcMAC,
+				dstMAC:      dstMAC,
+				metadata:    m,
+				useIpv6:     ipv6,
+				trafficPort: getExpectedTrafficPort(m),
+				packetIO:    getTracerouteParameter(t),
+			}
+
+			t.Run(args.testName(), func(t *testing.T) {
+				args.packetIO = getTracerouteParameter(t)
+				testPacketOut(ctx, t, args)
+			})
+		}
 	}
+}
+
+// generates all possible combinations of metadata
+// including omission
+func genMetadataCombinations() [][]*p4v1.PacketMetadata {
```

Comment:
this should just be a table of tests not a function so we know what subtests failed

Reply:
I'm not sure how that would be different. The test name will tell you what subtest has failed.

Reply:
To clarify, this is simply generating that table dynamically, rather than having to manually specify all 18 or so subtests.